src/c2goto/library/string.c

```
@@ -26,9 +26,21 @@
 char *strcpy(char *dst, const char *src)
 {
 __ESBMC_HIDE:;
-  char *cp = dst;
-  while ((*cp++ = *src++))
-    ;
+  // Ensure src pointer is non-null
+  if (src == NULL)
```

If src is null, we should IMHO crash/assert and not silently return dst.